util/fibers/detail/scheduler.cc

```diff
@@ -253,10 +253,6 @@ void Scheduler::ScheduleFromRemote(FiberInterface* cntx) {
     cntx->DEBUG_remote_epoch = remote_epoch_.fetch_add(1, memory_order_relaxed);
     remote_ready_queue_.Push(cntx);
 
-
-    // clear the bit after we pushed to the queue.
-    cntx->flags_.fetch_and(~FiberInterface::kScheduleRemote, memory_order_release);
```

Comment:
I find it slightly confusing that if `isScheduledRemotely()` is true above we clear the bit here whereas for this case we clear the bit when we pop the item in from the remote queue in the remote scheduler. What I had in mind is to clear the bit here and on the remote scheduler side to place back the item at the back of the remote_queue. That way this flow still remains in control of unseting the bit. 

IMO I don't see a clear benefit in my suggestion so your change good as well.